retrofit2-clients/src/main/java/com/palantir/remoting3/retrofit2/UnwrapRemoteIoExceptionCallAdapterFactory.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * (c) Copyright 2017 Palantir Technologies Inc. All rights reserved.
+ */
+
+package com.palantir.remoting3.retrofit2;
+
+import com.palantir.remoting.api.errors.RemoteException;
+import com.palantir.remoting3.okhttp.RemoteIoException;
+import java.io.IOException;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import javax.annotation.Nullable;
+import retrofit2.Call;
+import retrofit2.CallAdapter;
+import retrofit2.Callback;
+import retrofit2.Response;
+import retrofit2.Retrofit;
+
+/**
+ * This class ensures retrofit2 interfaces that return a plain {@code retrofit2.Call<T>} receive a
+ * proper {@link RemoteException} on both their synchronous and async codepaths instead of the
+ * {@link RemoteIoException} wrapper class.
+ * <p>
+ * Similar to the Java8 Competable Future logic in {@link AsyncSerializableErrorCallAdapterFactory}.
+ */
+public final class UnwrapRemoteIoExceptionCallAdapterFactory extends CallAdapter.Factory {
+    public static final CallAdapter.Factory INSTANCE = new UnwrapRemoteIoExceptionCallAdapterFactory();
+
+    @Nullable
+    @Override
+    public CallAdapter<?, ?> get(Type returnType, Annotation[] annotations, Retrofit retrofit) {
+        if (getRawType(returnType) != retrofit2.Call.class) {
```

Comment:
Call.class, not retrofit2.Call.class

Reply:
I've been using the fully qualified `retrofit2.Call` in a few places because while writing this feature, the code got very confusing because the `okhttp3.Call` and `retrofit2.Call` both abbreviate to the same thing.  This caused me significant grief because the `Retrofit.Builder#callFactory()` method uses the okhttp Call, whereas the `Retrofit.Builder#callAdapterFactory()` uses the retrofit2 Call!